### ISSUE TITLE
Cannot get mandatory column after doing an "optional" join

### ISSUE THREAD

**Issue by ansman** [BoD/android-contentprovider-generator#73]
We have these tables:
```json
// table1.json
{
  "fields": [{
    "name": "capture_id",
    "type": "Long",
    "nullable": true,
    "index": true,
    "foreignKey": {
        "table": "table2",
        "onDelete": "CASCADE"
    }
  }]
}

// table2.json
{
  "fields": [{
    "name": "name",
    "type": "String",
    "nullable": false
  }]
}
```

So when we do this:
```java
Table1Cursor cursor = new Table1Selection()
    .id(1)
    .query(context.getContentResolver(), new String[] {Table2.name, ...});
cursor.moveToNext();
cursor.getTable2Name(); // Crash because it throws a NullPointerException
```

It possible it would be great if we allowed non null columns for tables joined with nullable foreign keys to return null

**Comment by BoD** [comment]
I actually think it should work.  Can you paste the exact exception, with a stacktrace?

**Comment by ansman** [comment]
The problem is this code in `Table1Cursor`

```java
    /**
     * Get the {@code name} value.
     * Cannot be {@code null}.
     */
    @NonNull
    public String getTable2Name() {
        String res = getStringOrNull(Table2Columns.NAME);
        if (res == null)
            throw new NullPointerException("The value of 'name' in the database was null, which is not allowed according to the model definition");
        return res;
    }
```

Unfortunately I cannot post any actual code since it might contain sensitive information.

**Comment by BoD** [comment]
I understand the problem now, thanks :)
I will think about this, I hope I can come up with a solution.

**Comment by BoD** [comment]
Resolved in 0fded19
